### PR TITLE
refactor: add monaco editor default options and setGlobalEditorTheme api

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Refactor**
  - Simplified editor configuration by removing specific display settings such as theme, tab size, auto-indent, format on paste, automatic layout, and rounded selection.
  - Introduced global theme setting for the Monaco editor.

- **New Features**
  - Added a global Monaco editor theme configuration for more consistent theme management across the application.

### CODE REVIEW

File: packages/common/component/VueMonaco.vue [hunk 1]
```diff
@@ -173,7 +186,7 @@ export default {
     )
 
     watch(
-      () => props.theme,
+      () => globalEditorTheme,
```
Comment (hexqi) [inline packages/common/component/VueMonaco.vue]: globalEditorTheme是一个ref对象，地址不会变，watch ref变量要加.value

File: packages/common/index.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ import CloseIcon from './component/CloseIcon.vue'
 import LifeCycles from './component/LifeCycles.vue'
 import EmptyTip from './component/EmptyTip.vue'
 import MaskModal from './component/MaskModal.vue'
-import VueMonaco from './component/VueMonaco.vue'
+import VueMonaco, { setGlobalEditorTheme } from './component/VueMonaco.vue'
```
Comment (hexqi) [inline packages/common/index.js]: 导出方法名用`setGlobalMonacoEditorTheme`更明确一点

File: packages/common/component/VueMonaco.vue [hunk 3]
```diff
@@ -6,6 +6,15 @@ import * as monacoEditor from 'monaco-editor'
 import { watch, onMounted, nextTick, onBeforeUnmount, ref } from 'vue'
 import { formatString } from '@opentiny/tiny-engine-controller/js/ast'
 
+const globalEditorTheme = ref('vs')
+
+/**
+ * @param {'vs' | 'vs-dark'} theme
+ */
+export const setGlobalEditorTheme = (theme) => {
```
Comment (hexqi) [inline packages/common/component/VueMonaco.vue]: 参数theme可以加个默认值vs，不传恢复默认主题

File: packages/common/index.js (hunk 2 above)
Comment (gene9831) [reply packages/common/index.js]: optimized

File: packages/common/component/VueMonaco.vue (hunk 3 above)
Comment (gene9831) [reply packages/common/component/VueMonaco.vue]: optimized

File: packages/common/component/VueMonaco.vue (hunk 1 above)
Comment (gene9831) [reply packages/common/component/VueMonaco.vue]: update to `watch(refObject)` directly

**Review by hexqi — APPROVED** [review]

**Pull request merged**